app/code/Magento/SendFriendGraphQl/Model/Provider/GetVisibleProduct.php

```diff
@@ -49,8 +49,9 @@ public function execute(int $productId): ProductInterface
             $product = $this->productRepository->getById($productId);
 
             if (!in_array(
-                $product->getVisibility(),
-                $this->visibility->getVisibleInCatalogIds()
+                (int) $product->getVisibility(),
+                $this->visibility->getVisibleInSiteIds(),
```

Comment:
The docblock hint for this function's return value is incorrectly `string[]` proposed fix here https://github.com/magento/magento2/pull/23532